contracts/oracle/CollateralizationMemoizer.sol

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "./IOracle.sol";
+import "./ICollateralizationOracle.sol";
+import "../refs/CoreRef.sol";
```

Comment:
This contract should use `../utils/Timed.sol` to keep track of the validityDuration and lastUpdate. Is there a reason why you didn't?

Reply:
no reason, I just needed a timestamp, didn't feel like inheriting the whole class, but why not